### ISSUE TITLE
Training hangs at 33rd step of 434th epoch in a multinode distributed setting

### ISSUE THREAD

**Issue by rakshithvasudev** [NVIDIA/waveglow#160]
Hello,

I've been training waveglow from scratch (after training tacotron2) with 64 V100s using 16 nodes having slurm as a scheduler. 

This is how the submission script looks like:       
`python $TTS/train.py -m WaveGlow -o output_waveglow/ -lr 1e-4 --epochs 2001 -bs 4 --segment-length  8000 --weight-decay 0 --grad-clip-thresh 3.4028234663852886e+38 --cudnn-benchmark --cudnn-enabled --log-file output_waveglow/nvlog.json --rank $SLURM_PROCID --world-size $1 --dist-url $2`

Training works just fine until it hits this point: 

```
:::NVLOGv0.2.2 Tacotron2_PyT 1568948094.299715281 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) iter_time: 1.7069056034088135
Batch: 32/49 epoch 434
:::NVLOGv0.2.2 Tacotron2_PyT 1568948094.346622467 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iter_start: 32
:::NVLOGv0.2.2 Tacotron2_PyT 1568948094.923844576 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iteration_loss: -5.381126880645752
:::NVLOGv0.2.2 Tacotron2_PyT 1568948096.047455311 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iter_stop: 32
:::NVLOGv0.2.2 Tacotron2_PyT 1568948096.047880650 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iter_items/sec: 1203618.0827689199
:::NVLOGv0.2.2 Tacotron2_PyT 1568948096.048263073 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) iter_time: 1.7015364170074463
Batch: 33/49 epoch 434
:::NVLOGv0.2.2 Tacotron2_PyT 1568948096.066123247 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iter_start: 33
:::NVLOGv0.2.2 Tacotron2_PyT 1568948096.698991299 (/mnt/isilon/user/models/tt2/DeepLearningExamples/PyTorch/SpeechSynthesis/Tacotron2/dllogger/logger.py:251) train_iteration_loss: -5.253488063812256
```

and it hangs after this point. There are no logs, no errors, GPUs are being utilized across all nodes. I checked to see if I lost any process in each node, but all the processes are active. Meaning, there are 4 processes per node - (we have 4 GPUs per node). 

I tried running this training twice and both the time it was frozen at 33rd step of 434th epoch. Wonder what makes it be frozen? Is it doing anything special at that point in time? Is it frozen because it is waiting on something? 

This is how my env looks like:

```
Os: PRETTY_NAME="CentOS Linux 7 (Core)"

CUDA:
cuda10.0/toolkit/10.0.130/bin/nvcc

Sample nvdia-smi from one of the node:
Every 2.0s: nvidia-smi                                                                                                                                            Fri Sep 20 10:51:57 2019

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.67       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  On   | 00000000:18:00.0 Off |                    0 |
| N/A   49C    P0    74W / 300W |  12878MiB / 32480MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  On   | 00000000:3B:00.0 Off |                    0 |
| N/A   41C    P0    70W / 300W |  12928MiB / 32480MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  On   | 00000000:86:00.0 Off |                    0 |
| N/A   41C    P0    72W / 300W |  12920MiB / 32480MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  On   | 00000000:AF:00.0 Off |                    0 |
| N/A   48C    P0    72W / 300W |  12868MiB / 32480MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0    177499      C   python                                     12867MiB |
|    1    177494      C   python                                     12917MiB |
|    2    177493      C   python                                     12909MiB |
|    3    177495      C   python                                     12857MiB |
+-----------------------------------------------------------------------------+
```

Conda
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
asn1crypto                0.24.0                   py37_0
blas                      1.0                         mkl
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2019.5.15                     1
certifi                   2019.6.16                py37_1
cffi                      1.12.3           py37h2e261b9_0
chardet                   3.0.4                    py37_1
conda                     4.7.11                   py37_0
conda-package-handling    1.3.11                   py37_0
cryptography              2.7              py37h1ba5d50_0
cudatoolkit               10.0.130                      0
freetype                  2.9.1                h8a8886c_1
idna                      2.8                      py37_0
intel-openmp              2019.4                      243
jpeg                      9b                   h024ee3a_2
libarchive                3.3.3                h5d8350f_5
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libgfortran-ng            7.3.0                hdf63c60_0
libpng                    1.6.37               hbc83047_0
libstdcxx-ng              9.1.0                hdf63c60_0
libtiff                   4.0.10               h2733197_2
libxml2                   2.9.9                hea5a465_1
lz4-c                     1.8.1.2              h14c3975_0
lzo                       2.10                 h49e0be7_2
mkl                       2019.4                      243
mkl-service               2.3.0            py37he904b0f_0
mkl_fft                   1.0.14           py37ha843d7b_0
mkl_random                1.0.2            py37hd81dba3_0
ncurses                   6.1                  he6710b0_1
ninja                     1.9.0            py37hfd86e86_0
numpy                     1.16.4           py37h7e9f1db_0
numpy-base                1.16.4           py37hde5b4d6_0
olefile                   0.46                     py37_0
openssl                   1.1.1d               h7b6447c_1
pillow                    6.1.0            py37h34e0f95_0
pip                       19.1.1                   py37_0
pycosat                   0.6.3            py37h14c3975_0
pycparser                 2.19                     py37_0
pyopenssl                 19.0.0                   py37_0
pysocks                   1.7.0                    py37_0
python                    3.7.3                h0371630_0
python-libarchive-c       2.8                     py37_11
pytorch                   1.0.1           py3.7_cuda10.0.130_cudnn7.4.2_2    pytorch
readline                  7.0                  h7b6447c_5
requests                  2.22.0                   py37_0
ruamel_yaml               0.15.46          py37h14c3975_0
setuptools                41.0.1                   py37_0
six                       1.12.0                   py37_0
sqlite                    3.29.0               h7b6447c_0
tk                        8.6.8                hbc83047_0
torchvision               0.2.2                      py_3    pytorch
tqdm                      4.32.1                     py_0
urllib3                   1.24.2                   py37_0
wheel                     0.33.4                   py37_0
xz                        5.2.4                h14c3975_4
yaml                      0.1.7                had09818_2
zlib                      1.2.11               h7b6447c_3
zstd                      1.3.7                h0b5b093_0

```

Any help would be appreciated.

**Comment by rakshithvasudev** [comment]
I tried to continue to train by resuming from saved checkpoint. It resumes training without any problem, until it reaches the next 400 epochs, at which point it hangs. It is a deadlock issue.

This workaround fixed my issue : 
https://github.com/pytorch/pytorch/issues/20630#issuecomment-494959190
